apps/portal/src/app/contracts/modular-contracts/get-started/create-core-contract/page.mdx

```diff
@@ -29,152 +34,196 @@ forge remappings > remappings.txt
+```solidity
+ // SPDX-License-Identifier: MIT
+ pragma solidity ^0.8.20;
 
-   import {Core} from "modular-contracts/src/Core.sol";
+ import {Core} from "@thirdweb-dev/src/Core.sol";
+ import {BeforeIncrementCallback} from "./interface/BeforeIncrementCallback.sol";
 
-   contract CounterCore is Core {
+ contract CounterCore is Core {
+     constructor(address owner) {
+         _initializeOwner(owner);
+     }
+ }
 
-        constructor(address owner) {
-            _initializeOwner(owner);
-        }
+```
+
+> **Note**  
+> The `Core` contract is the base contract that needs to be inherited for this contract to be recognized as a core contract.
+
+</Step>
 
-   }
-   ```
+<Step title="Set Get Supported Callback Function">
+Implement the `getSupportedCallbackFunctions` function. The Core contract is abstract because this function is not implemented. To avoid compilation errors, declare the function with an empty body for now.
 
-   > **Note**  
-   > The `Core` contract is the base contract that needs to be inherited for this contract to be recognized as a core contract.
+    ```solidity
+        // SPDX-License-Identifier: MIT
+        pragma solidity ^0.8.20;
+
+        import {Core} from "@thirdweb-dev/src/Core.sol";
+
+        contract CounterCore is Core {
+            constructor(address owner) {
+                _initializeOwner(owner);
+            }
+
+            function getSupportedCallbackFunctions()
+                public
+                pure
+                override
+                returns (SupportedCallbackFunction[] memory supportedCallbackFunctions)
+            {}
+        }
+    ```
 
 </Step>
 
 <Step title="Set Up Increment Function">
    Define a function to increment a counter.
 
-   ```solidity
-   // SPDX-License-Identifier: UNLICENSED
-   pragma solidity ^0.8.20;
+```solidity
+ // SPDX-License-Identifier: MIT
+ pragma solidity ^0.8.20;
 
-   import {Core} from "modular-contracts/src/Core.sol";
+ import {Core} from "@thirdweb-dev/src/Core.sol";
 
-   contract CounterCore is Core {
-       uint256 public count;
+ contract CounterCore is Core {
+     uint256 public count;
 
-        constructor(address owner) {
-            _initializeOwner(owner);
-        }
+     constructor(address owner) {
+         _initializeOwner(owner);
+     }
+
+     function getSupportedCallbackFunctions()
+         public
+         pure
+         override
+         returns (SupportedCallbackFunction[] memory supportedCallbackFunctions)
+     {}
 
-       function increment() public {
-           count += 1;
-       }
-   }
-   ```
+     // 👇👇👇👇👇👇👇👇👇
+     function increment() public {
+         count += 1;
+     }
+ }
+```
 
 </Step>
 
 <Step title="Add a Callback Function">
-   Introduce the `_beforeIncrement` function to use the `beforeIncrement` callback from a module.
+   Introduce the `_beforeIncrement` function to use the `beforeIncrement` callback from a module to achieve this, we'll introduce the interface `BeforeIncrementCallback`
```

Comment:
Apologizes, what I meant was to split it up into 2 sentences so that it's not a run on sentence

```
Introduce the `_beforeIncrement` function to use the `beforeIncrement` callback from a module.
To achieve this, we'll introduce the interface `BeforeIncrementCallback`
```